code/modules/mob/living/carbon/xenomorph/XenoProcs.dm

```diff
@@ -547,3 +576,16 @@
 			return
 	ammo = ammo_list[spit_types[1]] //No matching projectile time; default to first spit type
 	return
+
+/mob/living/carbon/Xenomorph/proc/stealth_router(code = 0)
+	if(!istype(src,/mob/living/carbon/Xenomorph/Hunter))
```

Comment:
Make this something like:
```
/mob/living/carbon/Xenomorph/proc/stealth_router(code = 0)
	return FALSE

/mob/living/carbon/Xenomorph/Hunter/stealth_router(code = 0)
	[code]
```
Same effect, but better performance.
You can also keep `stealth` as a Hunter-only var that way.